lib/pure/strformat.nim

```diff
@@ -573,19 +587,24 @@ proc strformatImpl(pattern: NimNode; openChar, closeChar: char): NimNode =
           strlit = ""
 
         var subexpr = ""
-        while i < f.len and f[i] != closeChar and f[i] != ':':
-          if f[i] == '=':
+        var inParens = 0
+        while i < f.len and f[i] != closeChar and (f[i] != ':' or inParens!=0):
```

Comment:
`f[i] != ':'` is not consistent with `inParens!=0`.

Reply:
```if f[i] == ':' and inParens==0: break``` is the condition.
could be ```not (f[i]==':' and inParens==0)``` if that's clearer
 not(A and B) == (not A or not B)

Reply:
oh sorry i get what you meant. spacing